core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenTaskSource.java

```diff
@@ -464,15 +464,15 @@ private synchronized void processNext()
                 return;
             }
             verify(splitLoadingFuture == null || splitLoadingFuture.isDone(), "splitLoadingFuture is still running");
-            long start = System.currentTimeMillis();
+            long start = System.nanoTime();
             splitLoadingFuture = splitSource.getNextBatch(splitBatchSize);
             Futures.addCallback(splitLoadingFuture, new FutureCallback<>()
             {
                 @Override
                 public void onSuccess(SplitBatch result)
                 {
                     try {
-                        getSplitTimeRecorder.accept(System.currentTimeMillis() - start);
+                        getSplitTimeRecorder.accept(System.nanoTime() - start);
```

Comment:
Looking at `io.trino.execution.StageStateMachine#recordGetSplitTime` the param should be `startNanos` (variab le `start` in this case).
On a related topic, is it possible to test this?

Reply:
Right now there are no tests for that, but I'm going to look into the matter of testing this during https://github.com/trinodb/trino/pull/14713 task.